pkg/kubectl/cmd/get.go

```diff
@@ -178,6 +178,10 @@ func RunGet(f cmdutil.Factory, out, errOut io.Writer, cmd *cobra.Command, args [
 		return err
 	}
 
+	if strings.HasPrefix(cmdNamespace, "namespaces/") {
+		cmdNamespace = strings.Split(cmdNamespace, "/")[1]
```

Comment:
also, this should be done in `DefaultNamespace`, not in one specific command. the behavior of `-n ...` should be consistent